From 19622514c69e798d91e7908090b3d79dfdb14715 Mon Sep 17 00:00:00 2001
From: Mayank Grover <groverm@codeaurora.org>
Date: Tue, 24 Oct 2017 13:41:34 +0530
Subject: app: aboot: Zeroed out buffer, before flashing to prevent leak.

Zeroed out dev-info buffer to prevent leak of information.

Change-Id: I136afe9232f5fa6cbacb9948f5aa768cb9749008
---
 app/aboot/aboot.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/app/aboot/aboot.c b/app/aboot/aboot.c
index 95fdb4b..d552e44 100755
--- a/app/aboot/aboot.c
+++ b/app/aboot/aboot.c
@@ -2102,6 +2102,7 @@ void write_device_info_flash(device_info *dev)
 			return;
 	}
 
+	memset(info, 0, BOOT_IMG_MAX_PAGE_SIZE);
 	memcpy(info, dev, sizeof(device_info));
 
 	if (flash_write(ptn, 0, (void *)info_buf, page_size))
-- 
cgit v1.1

